go/vt/vtctl/workflow/server.go

```diff
@@ -1101,6 +1068,62 @@ func (s *Server) MoveTablesCreate(ctx context.Context, req *vtctldatapb.MoveTabl
 		return nil, err
 	}
 
+	// If we get an error after this point, where the vreplication streams/records
+	// have been created, then we clean up the workflow's artifacts.
+	defer func() {
+		if err != nil {
+			ts, cerr := s.buildTrafficSwitcher(ctx, ms.TargetKeyspace, ms.Workflow)
+			if cerr != nil {
+				err = vterrors.Wrapf(err, "failed to cleanup workflow artifacts: %v", cerr)
+			}
+			if cerr := s.dropArtifacts(ctx, false, &switcher{s: s, ts: ts}); cerr != nil {
+				err = vterrors.Wrapf(err, "failed to cleanup workflow artifacts: %v", cerr)
+			}
+		}
+	}()
+
+	// Now that the streams have been successfully created, let's put the associated
+	// routing rules in place.
+	if externalTopo == nil {
+		// Save routing rules before vschema. If we save vschema first, and routing
+		// rules fails to save, we may generate duplicate table errors.
```

Comment:
There are no cases where if routing rules succeed to be saved, but then the vschema fails that things are inconsistent?

Reply:
That's a good question! It led me to realize that we can instead utilize the `WorkflowDelete()` `VtCtldServer` function in the `defer()` as this performs additional cleanup, including the vschema.

Reply:
Turns out that wasn't quite right, but I did add code to deal with the vschema in the defer.

Reply:
Turns out that wasn't quite right, but I did add code to deal with the vschema in the defer: https://github.com/vitessio/vitess/pull/13737/commits/80f8a0314687169298f98466e040e92ab4e461ce